### PR TITLE
feat: callback_data auto serialization in InlineButton

### LINKED ISSUE #28: auto serialize callback_data and other composite types in keyboard interfaces
example:
```python
InlineButton("Dummy button", callback_data={"action": "buy", "quantity": 3})
```

### PR DESCRIPTION
**Pull request by VodoGamer**
closes #28

### CODE REVIEW

File: telegrinder/bot/rules/callback_data.py [hunk 1]
```diff
@@ -46,7 +47,7 @@ async def check(self, event: CallbackQuery, ctx: dict) -> bool:
 
 
 class CallbackDataJsonModel(CallbackQueryRule):
-    def __init__(self, model: typing.Type[msgspec.Struct]):
+    def __init__(self, model: typing.Type[msgspec.Struct] | IsDataclass):
```
Comment (luwqz1) [inline telegrinder/bot/rules/callback_data.py:50]: ```msgspec.json.Decoder``` parameter *type* expects the type and you hinted `IsDataclass`, correct to `type[IsDataclass]`.

**Comment by luwqz1** [comment]
cool

**Comment by luwqz1** [comment]
close #38

**Review by timoniq — APPROVED** [review]
Thank you!

**Pull request merged**